client/java/src/main/java/io/openlineage/client/transports/CompositeConfig.java

```diff
@@ -0,0 +1,94 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.openlineage.client.MergeConfig;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+/**
+ * CompositeConfig is a configuration class for CompositeTransport, implementing TransportConfig.
+ */
+@ToString
+@NoArgsConstructor
+public final class CompositeConfig implements TransportConfig, MergeConfig<CompositeConfig> {
+
+  @Getter @Setter private List<TransportConfig> transports;
+
+  @Getter @Setter private boolean continueOnFailure;
+
+  @JsonCreator
+  @SuppressWarnings("unchecked")
+  public CompositeConfig(
+      @JsonProperty("transports") Object transports,
+      @JsonProperty("continueOnFailure") boolean continueOnFailure) {
+
+    if (transports instanceof List) {
+      // Handle List<Map<String, Object>> case
```

Comment:
isn't it a list?

Reply:
So this is to handle both List and Map of Lists when instantianting with Jackson.
I don't like the idea of having to name each transport, you could just pass list of configurations with `name` optional. Also, currently Python version supports list only (to be changed soon when adding support for configuration through env vars).

I don't have strong opinion whether both List and List of Maps should be accepted.